cpp/include/algorithms.hpp

```diff
@@ -1280,5 +1280,29 @@ random_walks(raft::handle_t const &handle,
              typename graph_t::vertex_type const *ptr_d_start,
              index_t num_paths,
              index_t max_depth);
+
+/**
+ * @brief returns the COO format (src_vector, dst_vector) from the random walks (RW)
+ * paths.
+ *
+ * @tparam vertex_t Type of vertex indices.
+ * @tparam index_t Type used to store indexing and sizes.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param coalesced_sz_v coalesced vertex vector size.
+ * @param num_paths number of paths.
+ * @param d_coalesced_v coalesced vertex buffer.
+ * @param d_sizes paths size buffer.
+ * @return tuple of (src_vertex_vector, dst_Vertex_vector, path_offsets), where
+ * path_offsets are the offsets where the COO set of each path starts.
+ */
+template <typename vertex_t, typename index_t>
+std::
+  tuple<rmm::device_uvector<vertex_t>, rmm::device_uvector<vertex_t>, rmm::device_uvector<index_t>>
+  convert_paths_to_coo(raft::handle_t const &handle,
```

Comment:
I didn't really like this defined in `algorithms.hpp`, either. But `cython.*` files don't include `functions.hpp` (not directly, at least). And if they do not, then including `functions.hpp` in `cython.*` would add yet another dependency in already large `cython.*`, with increased compile times, etc. 

Also, `functions.hpp` seems legacy code on its way to being retired after code re-org. Perhaps, you meant `graph_functions.hpp`?

Reply:
`graph_functions.hpp` seems better than `algorithms.hpp`, I agree.  I hate to add it to `experimental` though, but maybe that's still better than `algorithms.hpp`?  Your call, I was just pointing out that if you had a better idea on where it could live, maybe now is a good time to move it before it gets settled in :)

Reply:
Hmm. `graph_functions.hpp` is dedicated to _generic_ graph functionality / helpers (@seunghwak , please correct me if I'm wrong). This function however is a dedicated helper (COO conversion) for a dedicated piece of functionality (RW). Also, `graph_functions.hpp` has definitions, not just declarations. I doubt that's the right place for it. I think `algorithms.hpp` is still better than it.

How about if I just _declare_ the function inside `cython.hpp` ? (we're only talking declaration here, anyway; the definition has always been in `random_walks.cuh`).

Reply:
Actually, probably better, I'll just put that declaration in its own header: `src/sampling/rw_functions.hpp` and include it in `cython.cu`.

Reply:
I recently added this function https://github.com/rapidsai/cugraph/blob/branch-0.20/cpp/src/utilities/path_retrieval.cu
We might want to put accelerated utility functions like this in a single place.

Reply:
@hlinsen , okay I see an `include/utilities/path_retrieval.hpp` with declarations only. I'll try including that in `cython.cu`. Thanks.

Reply:
Done.